azure-proto-core/ArmClient.cs

```diff
@@ -1,161 +1,109 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
-
+﻿using Azure;
+using Azure.Core;
+using Azure.Identity;
+using Azure.ResourceManager.Resources;
+using Azure.ResourceManager.Resources.Models;
+using azure_proto_core.Adapters;
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Runtime.CompilerServices;
 using System.Threading;
 using System.Threading.Tasks;
-using Azure;
-using Azure.Core;
-using Azure.Identity;
-using Azure.ResourceManager.Resources;
-using Azure.ResourceManager.Resources.Models;
-using azure_proto_core.Adapters;
 
 namespace azure_proto_core
 {
     /// <summary>
-    ///     The entry point for all ARM clients.  Note that, we may not want to take a dirrect dependency on Azure.Identity, so
-    ///     we may make the
-    ///     credential required.
-    ///     TODO: What is appropriate naming for ArmClient , given that we would not liek to make distinctions between data and
-    ///     management.
+    /// The entry point for all ARM clients.  Note that, we may not want to take a dirrect dependency on Azure.Identity, so we may make the
+    /// credential required.
+    /// TODO: What is appropriate naming for ArmClient , given that we would not liek to make distinctions between data and management.
     /// </summary>
     public class ArmClient
     {
         internal static readonly string DefaultUri = "https://management.azure.com";
 
-        public ArmClient()
-            : this(new Uri(DefaultUri), new DefaultAzureCredential(), null, new ArmClientOptions())
-        {
-        }
+        public Dictionary<string, string> ApiVersionOverrides { get; private set; }
 
-        public ArmClient(string defaultSubscriptionId)
-            : this(new Uri(DefaultUri), new DefaultAzureCredential(), defaultSubscriptionId, new ArmClientOptions())
-        {
-        }
+        public ArmClient(ArmClientOptions options = null)
+            : this(new Uri(DefaultUri), new DefaultAzureCredential(), null, options ?? new ArmClientOptions()) { }
 
-        public ArmClient(TokenCredential credential, string defaultSubscriptionId)
-            : this(new Uri(DefaultUri), credential, defaultSubscriptionId, new ArmClientOptions())
-        {
-        }
+        public ArmClient(string defaultSubscriptionId, ArmClientOptions options = null)
+            : this(new Uri(DefaultUri), new DefaultAzureCredential(), defaultSubscriptionId, options ?? new ArmClientOptions()) { }
 
-        public ArmClient(Uri baseUri, TokenCredential credential)
-            : this(baseUri, credential, null, new ArmClientOptions())
-        {
-        }
+        public ArmClient(TokenCredential credential, string defaultSubscriptionId, ArmClientOptions options = null)
+            : this(new Uri(DefaultUri), credential, defaultSubscriptionId, options ?? new ArmClientOptions()) { }
```

Comment:
>options ?? new ArmClientOptions() [](start = 75, length = 33)

Don't need to have all these checks. Only need it at the lowest level of ctor and do it once.